homeassistant/components/recorder/__init__.py

```diff
@@ -788,34 +805,60 @@ def _run_external_statistics(self, metadata, stats):
         # Schedule a new statistics task if this one didn't finish
         self.queue.put(ExternalStatisticsTask(metadata, stats))
 
-    def _process_one_event(self, event):
+    def _lock_database(self, task: DatabaseLockTask):
+        @callback
+        def _async_set_database_locked(task: DatabaseLockTask):
+            task.database_locked.set()
+
+        with write_lock_db(self):
+            # Notify that lock is being held, wait until database can be used again.
+            self.hass.add_job(_async_set_database_locked, task)
+            while not task.database_unlock.wait(timeout=DB_LOCK_QUEUE_CHECK_TIMEOUT):
+                if self.queue.qsize() > MAX_QUEUE_BACKLOG * 0.9:
```

Comment:
```suggestion
                if self.queue.qsize() > MAX_QUEUE_BACKLOG * 0.9:
```

Maybe make `MAX_QUEUE_BACKLOG * 0.9`  a constant?

Reply:
Since the message just below refers to the 90% I like to have that factor nearby.